src/Ng/Jobs/Catalog2RegistrationJob.cs

```diff
@@ -78,12 +80,24 @@ protected override void Init(IDictionary<string, string> arguments, Cancellation
                 source,
                 storageFactories.LegacyStorageFactory);
 
-            RegistrationMakerCatalogItem.PackagePathProvider = new PackagesFolderPackagePathProvider();
+            if (isContentFlatContainer)
+            {
+                var flatContainerCursorUriString = arguments.GetOrThrow<string>(Arguments.FlatContainerCursorUri);
```

Comment:
The cursor is calculated now, but the flatcontainer uri is still passed as argument. I preferred this way instead of passing the index.json uri , open and read the flat container from there.